pkg/services/slack.go

```diff
@@ -14,8 +14,12 @@ import (
 
 	log "github.com/sirupsen/logrus"
 	"github.com/slack-go/slack"
+	"golang.org/x/time/rate"
 )
 
+// No rate limit unless Slack requests it (allows for Slack to control bursting)
+var rateLimiter = rate.NewLimiter(rate.Inf, 1)
```

Comment:
Why do you use global env?

Reply:
The controller can run N goroutines to process work items, so by using a global they all share the same rate limiter because from Slack's api, they would all be the same application.